src/App.js

```diff
@@ -1,13 +1,61 @@
 import React from 'react';
-
+import { GoodsList } from './components/GoodsList';
+import { getAll, get5First, getRed } from './api/goods';
 import './App.scss';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goods: null,
+    goodsIsSelected: false,
+  }
+
+  showGoods = (callback) => {
+    callback()
+      .then((goods) => {
+        this.setState({ goods });
+      })
+      .then(() => this.setState({ goodsIsSelected: true }));
```

Comment:
setState is not async, so this can be simplified:
```suggestion
      .then((goods) => {
        this.setState({
          goods,
          goodsIsSelected: true, 
        });
      });
```